Docs: Add Vue 2 integrations (#17117)
ckeditor/ckeditor5: docs/getting-started/integrations-cdn/vuejs-v2.md

---
menu-title: Vue.js 2.x
meta-title: Vue.js 2.x rich text editor component with CDN | CKEditor 5 documentation
meta-description: Install, integrate and configure CKEditor 5 using the Vue.js 2.x component with CDN.
category: cloud
order: 40
---

{@snippet installation/integrations/framework-integration}

# Vue.js 2.x rich text editor component

<p>
	<a href="https://www.npmjs.com/package/@ckeditor/ckeditor5-vue2" target="_blank" rel="noopener">
		<img src="https://badge.fury.io/js/%40ckeditor%2Fckeditor5-vue2.svg" alt="npm version" loading="lazy">
	</a>
</p>

<info-box warning>
	This guide is about the CKEditor&nbsp;5 integration with Vue.js 2.x. However, Vue 2 has reached EOL and is no longer actively maintained. To learn more about the integration with Vue.js 3+, check out the {@link getting-started/integrations-cdn/vuejs-v3"Rich text editor component for Vue.js 3+} guide.
</info-box>

> Vue.js is a versatile framework for building web user interfaces. CKEditor&nbsp;5 provides an official Vue component you can use in your application. This guide assumes that you already have a Vue 2 project.

Reviewer: ```suggestion
CKEditor 5 has an official Vue 2 integration that you can use to add a rich text editor to your application. This guide will help you install it and configure to use the CDN distribution of the CKEditor 5.
```

I think that we don't need to tell devs what Vue is. If they are on this page, then they very likely already know.

Author: Yeah, you're right. But I'm a little hesitant about using the "official integration" phrase in this case.

First, install the CKEditor&nbsp;5 WYSIWYG editor component for Vue 2:

```bash
npm install @ckeditor/ckeditor5-vue2
```

Then, include the CKEditor&nbsp;5 scripts and styles. All necessary scripts and links are in the HTML snippet below. You can copy and paste them into your `index.html`. Open-source and premium features are in separate files, so there are different tags for both types of plugins. Add tags for premium features only if you use them.

```html
<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<link rel="icon" href="/favicon.ico" />
		<meta name="viewport" content="width=device-width, initial-scale=1.0" />
		
		<link rel="stylesheet" href="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/ckeditor5.css" />
		<script src="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/ckeditor5.umd.js"></script>
		<!-- Add if you use premium features -->
        <link rel="stylesheet" href="https://cdn.ckeditor.com/ckeditor5-premium-features/{@var ckeditor5-version}/ckeditor5-premium-features.css" />
        <script src="https://cdn.ckeditor.com/ckeditor5/{@var ckeditor5-version}/ckeditor5-premium-features.umd.js"></script>
		<!--  -->

> 		<title>Vite App</title>

Reviewer 2: Vue?

Author: It's a bundler-generated title, which is Vite. But, sure - I'll change it.

Author: Hmm... It is a good point. On the other hand - it doesn't change much. So, I updated this detail.

	</head>
	<body>
		<div id="app"></div>
		<script type="module" src="/src/main.js"></script>
	</body>
</html>

```

To create an editor instance, you must first import the editor and the component modules into the root file of your application (for example, `main.js` when generated by create-vue).

```js
import Vue from 'vue';
import CKEditor from '@ckeditor/ckeditor5-vue2';
import App from './App.vue';

Vue.use( CKEditor );

> new Vue( { render: ( h ) => h( App ) } ).$mount( '#app' );

Reviewer: ```suggestion
new Vue( App ).$mount( '#app' );
```

```

Use the `<ckeditor>` component inside the template tag. The below example shows how to use the component with open-source plugins.

```html
<template>

> 	<div id="app">
> 		<ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
> 	</div>

Reviewer: ```suggestion
		<ckeditor :editor="editor" v-model="editorData" :config="editorConfig" />
```

</template>

<script>
const {
	ClassicEditor,
	Essentials,
	Bold,
	Italic,
	Paragraph
} = CKEDITOR;

export default {
	name: 'app',
	data() {
		return {
			editor: ClassicEditor,
			editorData: '<p>Hello from CKEditor 5 in Vue 2!</p>',
			editorConfig: {
				licenseKey: 'GPL',
				plugins: [ Bold, Essentials, Italic, Paragraph ],
				toolbar: [ 'undo', 'redo', '|', 'bold', 'italic' ]
			}
		};
	}
};
</script>
```

> The following example shows how to use the component with premium plugins. To make it work, you need to provide a proper license key.

Author: I deliberately divided the snippet into two. The snippet with premium plugins caused errors (globals not available). The open-source works after a simple copy-paste.

Author: The guide doesn't appear so poor, so that's another plus, I guess.

```html
<template>

> 	<div id="app">
> 		<ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
> 	</div>

Reviewer: ```suggestion
	<ckeditor :editor="editor" v-model="editorData" :config="editorConfig" />
```

</template>

<script>
const {
	ClassicEditor,
	Essentials,
	Bold,
	Italic,
	Paragraph,
	Mention
} = CKEDITOR;
const { SlashCommand } = CKEDITOR_PREMIUM_FEATURES;

export default {
	name: 'app',
	data() {
		return {
			editor: ClassicEditor,
			editorData: '<p>Hello from CKEditor 5 in Vue 2!</p>',
			editorConfig: {

> 				licenseKey: '<YOUR_LICENSE_KEY>',

Author: I deliberately didn't add "Or GPL" here. I divided the snippet into two - for open-source and premium plugins. So, the license key is the only option here.

Reviewer 2: Alright, sorry, I missed that part.

Author: No problem 👍

Vue.use( CKEditor );

new Vue( App ).$mount( '#app' );
```

Use the `<ckeditor>` component inside the template tag. The below example shows how to use the component with open-source and premium plugins. To make it work, you need to provide a proper license key.

```html
<template>
	<ckeditor :editor="editor" v-model="editorData" :config="editorConfig" />
</template>

<script>
const {
	ClassicEditor,
	Essentials,
	Bold,
	Italic,
	Paragraph,
	Mention
} = CKEDITOR;
const { SlashCommand } = CKEDITOR_PREMIUM_FEATURES;

export default {
	name: 'app',
	data() {
		return {
			editor: ClassicEditor,
			editorData: '<p>Hello from CKEditor 5 in Vue 2!</p>',
			editorConfig: {
				licenseKey: '<YOUR_LICENSE_KEY>', // Or 'GPL'.
				plugins: [ Bold, Essentials, Italic, Mention, Paragraph, SlashCommand ],
				toolbar: [ 'undo', 'redo', '|', 'bold', 'italic' ],
				mention: { 
					// Mention configuration
				}
			}
		};
	}
};
</script>
```

## Component directives

### `editor`

This directive specifies the editor to be used by the component. It must directly reference the editor constructor to be used in the template.

```html
<template>
	<ckeditor :editor="editor" />
</template>

<script>

> 	import { ClassicEditor } from 'ckeditor5';

Reviewer: ```suggestion
	const { ClassicEditor } = CKEDITOR;
```

Remember to do the same for other examples
